eng/packaging.targets

```diff
@@ -54,29 +76,55 @@
+      <TfmRuntimeSpecificPackageFile Include="$(_documentationFileToPackage)"
+                                     Condition="'$(TargetsAnyOS)' != 'true'" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="AddRuntimeSpecificFilesToPackage"
+          DependsOnTargets="BuiltProjectOutputGroup;
+                            SatelliteDllsProjectOutputGroup;
+                            $(TargetsForTfmSpecificBuildOutput);
+                            AddDocumentationFileToPackage"
+          Condition="'$(TargetsAnyOS)' != 'true'">
     <ItemGroup>
-      <TfmSpecificPackageFile Include="@(SatelliteDllsProjectOutputGroupOutput);
-                            @(BuiltProjectOutputGroupOutput)">
-        <PackagePath>runtimes/$(_runtimeOS)/lib/$(TargetFrameworkWithoutSuffix)</PackagePath>
-      </TfmSpecificPackageFile>
+      <TfmRuntimeSpecificPackageFile Include="@(BuiltProjectOutputGroupOutput);
+                                              @(SatelliteDllsProjectOutputGroupOutput)" />
+      <TfmSpecificPackageFile Include="@(TfmRuntimeSpecificPackageFile)"
+                              PackagePath="runtimes/$(PackageTargetRuntime)/lib/$(TargetFrameworkWithoutSuffix)" />
+      <!-- Copy runtime specific assemblies into the rid agnostic 'lib' folder if the ridless tfm doesn't exist.
+           The assumption holds that a ridless tfm doesn't follow a '-' charater. -->
+      <TfmSpecificPackageFile Include="@(TfmRuntimeSpecificPackageFile)"
+                              PackagePath="$([MSBuild]::ValueOrDefault('$(BuildOutputTargetFolder)', 'lib'))/$(TargetFrameworkWithoutSuffix)"
+                              Condition="!$([System.Text.RegularExpressions.Regex]::IsMatch('$(TargetFrameworks)', '$(TargetFrameworkWithoutSuffix)(?!-)'))" />
     </ItemGroup>
   </Target>
 
-  <!-- Runtime independent symbols are automatically added by the pack task.-->
-  <Target Name="AddRuntimeDebugSymbolsWithTfm"
-          Condition="'$(IncludeSymbols)' == 'true' and '$(TargetsAnyOS)' != 'true' and '$(ExcludeFromPackage)' == ''">
+  <!-- Runtime agnostic symbols are automatically added by the pack task.-->
+  <Target Name="AddRuntimeSpecificSymbolToPackage"
+          Condition="'$(IncludeSymbols)' == 'true' and
+                     '$(TargetsAnyOS)' != 'true'">
     <PropertyGroup>
-      <RuntimeSymbolPath>$([System.IO.Path]::GetDirectoryName($(TargetPath)))\$(TargetName).pdb</RuntimeSymbolPath>
+      <RuntimeSymbolPath>$(TargetDir)$(TargetName).pdb</RuntimeSymbolPath>
     </PropertyGroup>
 
-    <ItemGroup Condition="Exists('$(RuntimeSymbolPath)')">
-      <TfmSpecificDebugSymbolsFile Include="$(RuntimeSymbolPath)">
-        <TargetPath>/runtimes/$(_runtimeOS)/lib/$(TargetFrameworkWithoutSuffix)/$(TargetName).pdb</TargetPath>
-        <TargetFramework>$(TargetFrameworkWithoutSuffix)</TargetFramework>
-      </TfmSpecificDebugSymbolsFile>
+    <ItemGroup>
+      <TfmRuntimeSpecificDebugSymbolFile Include="$(RuntimeSymbolPath)"
```

Comment:
why do we need this extra item ?

Reply:
This serves as an extension point so that projects can add to this item when necessary. There's one example where this was necessary for the non symbol runtime specific assemblies.